idaes/generic_models/properties/core/examples/CustomMethods2_Final.py

```diff
@@ -0,0 +1,181 @@
+    get_method, get_component_object as cobj)
+from idaes.generic_models.properties.core.examples.CustomMethods_Final import \
+    vol_mol_liq_comp
+
+from idaes.core.util.misc import set_param_from_config
+
+# Molar Enthalpy of CO2 dissolved in liquid phase (J/mol)
+class enth_mol_liq_comp():
+
+    @staticmethod
+    def build_parameters(cobj):
+        NIST.cp_mol_ig_comp.build_parameters(cobj)
+
+    @staticmethod
+    def return_expression(b, cobj, T):
+        
+        # Specific enthalpy via the Shomate equation
+        t = pyunits.convert(T, to_units=pyunits.kiloK)
+        
+        h_form = b.params.config.include_enthalpy_of_formation
+        H = (cobj.cp_mol_ig_comp_coeff_H if not h_form
+              else 0*pyunits.kJ*pyunits.mol**-1)
+
+        h = (cobj.cp_mol_ig_comp_coeff_A*(t) +
+              (cobj.cp_mol_ig_comp_coeff_B/2)*(t**2) +
+              (cobj.cp_mol_ig_comp_coeff_C/3)*(t**3) +
+              (cobj.cp_mol_ig_comp_coeff_D/4)*(t**4) -
+              cobj.cp_mol_ig_comp_coeff_E*(1/t) +
+              cobj.cp_mol_ig_comp_coeff_F -
+              H)
+
+        units = b.params.get_metadata().derived_units
+        h1 = pyunits.convert(h, units["energy_mole"])
+        
+        # Hliq(T) = Hig(T) + R*T*ln(H_CO2_H2O*xco2/Pref)
+        # Data source [4], Eqn 13-33
+        htot = h1 + 8.314*(pyunits.J/(pyunits.mol*pyunits.K))*T*\
+            log(henrys_constant_co2_h2o.return_expression(b,cobj,T)*\
+                b.mole_frac_phase_comp['Liq', 'CO2']/\
+                (b.params.pressure_ref))
+        
+        return htot
+    
+class pressure_sat_comp():
+    
+    @staticmethod
+    def build_parameters(cobj):
+        pass
+    
+    @staticmethod
+    def return_expression(b, cobj, T, dT=False):
+        return 0
+
+# Henrys constant of CO2 dissolved in liquid phase (Pa.m3/mol)
+# Data source [3], N2O Analogy 
+class henrys_constant():
+    
+    @staticmethod
+    def build_parameters(cobj):
+        pass
+    
+    @staticmethod
+    def return_expression(b, cobj, T, xmea, xh2o, xco2):
+        T = pyunits.convert(T, to_units=pyunits.K)
+        t      = T - (273.15*pyunits.K)
+        den = (xmea*61.0831E-3*(pyunits.kg/pyunits.mol)) + \
+            (xh2o*18.0153E-3*(pyunits.kg/pyunits.mol))
+        wt_MEA = xmea*61.0831E-3*(pyunits.kg/pyunits.mol)/den
+        wt_H2O = xh2o*18.0153E-3*(pyunits.kg/pyunits.mol)/den
+        H_N2O_MEA = 2.448e5*exp(-1348*pyunits.K/T)
+        H_CO2_H2O = 3.52e6*exp(-2113*pyunits.K/T)
+        H_N2O_H2O = 8.449e6*exp(-2283*pyunits.K/T)
+        H_CO2_MEA = H_N2O_MEA*(H_CO2_H2O/H_N2O_H2O)
+        lwm = 1.70981 + 0.03972*(1/pyunits.K)*t - \
+            4.3e-4*(1/pyunits.K**2)*t**2 - 2.20377*wt_H2O
+        
+        return (exp(wt_MEA*log(H_CO2_MEA) +wt_H2O*log(H_CO2_H2O)
+                    + wt_MEA*wt_H2O*lwm))*pyunits.Pa*pyunits.m**3/pyunits.mol
+
+# Henrys constant of CO2 in H2O (Pa/mole fraction)
+# Data source [4], table 6.2-7
+class henrys_constant_co2_h2o():
```

Comment:
There will be a PR soon that adds support for Henry's components, so we should hold on on this PR until that is ready.